### ISSUE TITLE
1j01/diverge: 🔱 explore textual possibilities like never before

### ISSUE THREAD

**Issue by tg-z** [tg-z/web-clip#87]
1j01/diverge: 🔱 explore textual possibilities like never before                                       

![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA8AAAAPAgMAAABGuH3ZAAAAAXNSR0IArs4c6QAAAAlQTFRF/2YA////AAAALA0mBgAAAAFiS0dEAIgFHUgAAAAJcEhZcwAAEnMAABJzAYwiuQcAAAAHdElNRQfcCxUTCRv8+I/EAAAAMklEQVQI12NYtWrVCoapoaERDBMYGCSghAMLkBAQABIsDkCCkQHGAouBZeGKwXpBpgAAhtMRBKRIycIAAAAASUVORK5CYII=)
[![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFAAAAAPBAMAAACB51W8AAAAAXNSR0IArs4c6QAAACRQTFRFAAEAAxx+JCYjYWNgho53kJKPy6Gio63Ys7Wysb3lyMnG/f/8AJKv4wAAAAFiS0dEAIgFHUgAAAAJcEhZcwAACxMAAAsTAQCanBgAAAAHdElNRQfcBhsNKDWiRqoQAAAA2UlEQVQoz42SoQ7CMBCGL10w7CWWEAwzNRiwOEzD5DwvgSGoJVUoBAZFSGeWQZPSeznabutYtiX709b0y3/X/wowVSFW0vZ4iSMOq2hAGVgux2d98dl15UFmQMVr4zHwAAsFsMXMAJdltKKzYVACgTVAbA2RucYdaLZfDtSJhJgGPMPvHRk5QwKpvWvXvi5NjZ3pkeP1hDQoiaxAb9eAb0gtqDneULMW7DmWplb8gI19rOiC3R4VkCgsYW5T539gPx7lhmITFH4So4G7qM1kpoBC5DiqYvLn+QGXavuS25r9yAAAAABJRU5ErkJggg==)
](https://github.com/1j01/diverge/commits/master.atom "https://github.com/1j01/diverge/commits/master.atom")

[Skip to content](#start-of-content)

    ![](https://github.githubassets.com/images/search-key-slash.svg)

*   ![]()
    
    In this repository All GitHub ↵
    
    Jump to ↵
    

*   No suggested jump to results

*   ![]()
    
    In this repository All GitHub ↵
    
    Jump to ↵
    
*   ![]()
    
    In this repository All GitHub ↵
    
    Jump to ↵
    
*   ![](https://github.githubassets.com/images/spinners/octocat-spinner-128.gif)
    

[Pull requests](/pulls) [Issues](/issues)

[Marketplace](/marketplace)

[Explore](/explore)

[New repository](/new) [Import repository](/new/import) [New gist](https://gist.github.com/) [New organization](/organizations/new)

This repository

[New issue](/1j01/diverge/issues/new)

 ![](https://avatars0.githubusercontent.com/u/47585455?s=40&v=4) [Signed in as **tg-z**](/tg-z)

![](https://github.githubassets.com/images/icons/emoji/octocat.png)

lost

### Edit status

 

![](https://github.githubassets.com/images/icons/emoji/octocat.png)

![](https://github.githubassets.com/images/icons/emoji/octocat.png)

  

Could not update your status, please try again.

80 remaining

#### Suggestions:

🌴

On vacation

🤒

Out sick

🏠

Working from home

🎯

Focusing

 Busy 

When others mention you, assign you, or request your review, GitHub will let them know that you have limited availability.

Clear status

Never

*   Never
    
    Keep this status until you clear your status or edit your status.
    

*   in 30 minutes
*   in 1 hour
*   in 4 hours
*   today
*   this week

Set status Clear status

[Your profile](/tg-z) [Your repositories](/tg-z?tab=repositories) [Your projects](/tg-z?tab=projects) [Your stars](/tg-z?tab=stars) [Your gists](https://gist.github.com/mine)

[Help](https://help.github.com) [Settings](/settings/profile)

 Sign out 

*   _done_ In Queue
    
*   _bookmark_ Bookmarked
    

*         Watch   
    
    Notifications
    
    Not watching Be notified only when participating or @mentioned. Watch
    
    Releases only Be notified of new releases, and when participating or @mentioned. Unwatch releases
    
    Watching Be notified of all conversations. Unwatch
    
    Ignoring Never be notified. Stop ignoring
    
    [1](/1j01/diverge/watchers)
    
*       Unstar  [4](/1j01/diverge/stargazers) 
    
        Star  [4](/1j01/diverge/stargazers) 
    
*      Fork
    
    [0](/1j01/diverge/network/members)

[1j01](/1j01)/**[diverge](/1j01/diverge)**
==========================================

[Code  ](/1j01/diverge)   [  Issues 0  ](/1j01/diverge/issues)   [  Pull requests 0  ](/1j01/diverge/pulls)  [  Projects 0 ](/1j01/diverge/projects) [  Wiki ](/1j01/diverge/wiki) [  Security ](/1j01/diverge/security/advisories) [Insights](/1j01/diverge/pulse) 

🔱 explore textual possibilities like never before [https://1j01.github.io/diverge/](https://1j01.github.io/diverge/)

[text](/topics/text "Topic: text") [text-editor](/topics/text-editor "Topic: text-editor") [text-based](/topics/text-based "Topic: text-based") [exploration](/topics/exploration "Topic: exploration") [explorer](/topics/explorer "Topic: explorer") [explore](/topics/explore "Topic: explore") [exploratory](/topics/exploratory "Topic: exploratory") [words](/topics/words "Topic: words") [sentence](/topics/sentence "Topic: sentence") [sentences](/topics/sentences "Topic: sentences") [experiment](/topics/experiment "Topic: experiment") [experimental](/topics/experimental "Topic: experimental") [writing](/topics/writing "Topic: writing") [memes](/topics/memes "Topic: memes") [memex](/topics/memex "Topic: memex") [letters](/topics/letters "Topic: letters") [paths](/topics/paths "Topic: paths") [idea](/topics/idea "Topic: idea") [ideation](/topics/ideation "Topic: ideation") [wording](/topics/wording "Topic: wording")

*   [69 commits](/1j01/diverge/commits/master)
*   [3 branches](/1j01/diverge/branches)
*   [0 releases](/1j01/diverge/releases)
*   [1 contributor](/1j01/diverge/graphs/contributors)
*   [MIT](/1j01/diverge/blob/master/LICENSE.txt)

JavaScript HTML

1.  [JavaScript 95.4%](/1j01/diverge/search?l=javascript)
2.  [HTML 4.6%](/1j01/diverge/search?l=html)

_Branch:_ master [New pull request](/1j01/diverge/pull/new/master)

 Create new file 

[Upload files](/1j01/diverge/upload/master) [Find File](/1j01/diverge/find/master)

Clone or download

Use SSH

#### Clone with HTTPS[](https://help.github.com/articles/which-remote-url-should-i-use "Which remote URL should I use?")

Use Git or checkout with SVN using the web URL.

Use HTTPS

#### Clone with SSH[](https://help.github.com/articles/which-remote-url-should-i-use "Which remote URL should I use?")

You don't have any public SSH keys in your GitHub account. You can [add a new public key](/settings/ssh/new), or try cloning this repository via HTTPS.

Use a password protected SSH key.

[Open in Desktop](x-github-client://openRepo/https://github.com/1j01/diverge) [Download ZIP](/1j01/diverge/archive/master.zip)

#### Launching GitHub Desktop...

If nothing happens, [download GitHub Desktop](https://desktop.github.com/) and try again.

Go back

#### Launching GitHub Desktop...

If nothing happens, [download GitHub Desktop](https://desktop.github.com/) and try again.

Go back

#### Launching Xcode...

If nothing happens, [download Xcode](https://developer.apple.com/xcode/) and try again.

Go back

#### Launching Visual Studio...

If nothing happens, [download the GitHub extension for Visual Studio](https://visualstudio.github.com/) and try again.

Go back

[Gitpod](https://gitpod.io#https://github.com/1j01/diverge "Gitpod")

 [![](https://avatars0.githubusercontent.com/u/3630663?s=60&v=4)](/1j01) 

[1j01](/1j01/diverge/commits?author=1j01) [Check this out](/1j01/diverge/commit/d6ee97d4755ed423c60ab1af96f6e96a824f9a28 "Check this out")

Latest commit [d6ee97d](/1j01/diverge/commit/d6ee97d4755ed423c60ab1af96f6e96a824f9a28) on Apr 22

[Permalink](/1j01/diverge/tree/d6ee97d4755ed423c60ab1af96f6e96a824f9a28)
| Type | Name | Latest commit message | Commit time |
| --- | --- | --- | --- |
|  | Failed to load latest commit information. |
| ![](https://github.githubassets.com/images/spinners/octocat-spinner-32.gif)
 | [lib](/1j01/diverge/tree/master/lib "lib") | [This is mildly interesting](/1j01/diverge/commit/999c282dc2d2927bd5e4636fb66014670b98ecf9 "This is mildly interesting") | 3 years ago |
| ![](https://github.githubassets.com/images/spinners/octocat-spinner-32.gif)
 | [public](/1j01/diverge/tree/master/public "public") | [Inline CSS (fix flash of unstyled UI)](/1j01/diverge/commit/ff3e066bd4e8340b71f02b50e071ae62fd69f6aa "Inline CSS (fix flash of unstyled UI)") | 5 months ago |
| ![](https://github.githubassets.com/images/spinners/octocat-spinner-32.gif)
 | [src](/1j01/diverge/tree/master/src "src") | [Check this out](/1j01/diverge/commit/d6ee97d4755ed423c60ab1af96f6e96a824f9a28 "Check this out") | 5 months ago |
| ![](https://github.githubassets.com/images/spinners/octocat-spinner-32.gif)
 | [.gitattributes](/1j01/diverge/blob/master/.gitattributes ".gitattributes") |  [![](https://github.githubassets.com/images/icons/emoji/octocat.png)
 Added .gitattributes](/1j01/diverge/commit/08c6e79418c873ee49aace65693624aeb67eecb9 ":octocat: Added .gitattributes")  | 3 years ago |
| ![](https://github.githubassets.com/images/spinners/octocat-spinner-32.gif)
 | [.gitignore](/1j01/diverge/blob/master/.gitignore ".gitignore") | [Use create-react-app](/1j01/diverge/commit/0f4f159730b4ecda8e22b64e03d915e9fc9a0ba7 "Use create-react-app

Ran create-react-app CLI, merged gitignores, readmes, etc. and created a logo (using it as a favicon).") | 6 months ago |
| ![](https://github.githubassets.com/images/spinners/octocat-spinner-32.gif)
 | [LICENSE.txt](/1j01/diverge/blob/master/LICENSE.txt "LICENSE.txt") | [Update readme and add license](/1j01/diverge/commit/df60f666249216322ffc9985e2cb9116f63b7d5e "Update readme and add license") | 5 months ago |
| ![](https://github.githubassets.com/images/spinners/octocat-spinner-32.gif)
 | [README.md](/1j01/diverge/blob/master/README.md "README.md") | [Add most basic tab autocomplete](/1j01/diverge/commit/b622bd78bb4e58b31e6a0b5b38a8449f3fc2d611 "Add most basic tab autocomplete") | 5 months ago |
| ![](https://github.githubassets.com/images/spinners/octocat-spinner-32.gif)
 | [package.json](/1j01/diverge/blob/master/package.json "package.json") | [Clean up unused modules](/1j01/diverge/commit/33bf7f15a5df5e2c0d987a645625dc10e6d0fcb8 "Clean up unused modules") | 5 months ago |
| ![](https://github.githubassets.com/images/spinners/octocat-spinner-32.gif)
 | [yarn.lock](/1j01/diverge/blob/master/yarn.lock "yarn.lock") | [Clean up unused modules](/1j01/diverge/commit/33bf7f15a5df5e2c0d987a645625dc10e6d0fcb8 "Clean up unused modules") | 5 months ago |

### [README.md](http://README.md "Linkify Plus Plus")

[](#diverge)Diverge
===================

The medium is the message.

Diverge at any and every letter.

**This project is mostly just an idea so far.**

But [check out the beginnings.](https://1j01.github.io/diverge/)

### [](#paths)Paths

You'll be able to forge a path by typing, or follow suggested paths via mouse/pen/touch/whatever (or Tab on the keyboard).

Currently you can only use the keyboard, and to follow a suggested path, you have to type it out.

Similar paths that don't quite match could be lined up "magnetically"

Here's some crude ASCII-art to try to communicate the idea:

```
The quick fox jumped at the opportunity
The qui   fox jumps     the
       c n         over     lazy dog
      k   w
       bro

```

### [](#providers)Providers

Providers can look at the path you're on and the cursor position and suggest new paths.

The basic provider returns paths from a database of strings entered by users.

A Markov text chain or sentence autocomplete provider would return paths with text inserted at the cursor, whereas a paraphrasing provider (possibly using a neural network?) could return strings with infix replacements, making for paths that diverge and rejoin the current path.

### [](#evaluators)Evaluators

Do you value rhyme, alliteration, assonance? Are fifthglyphs [vil**e** & **e**vil](https://www.reddit.com/r/AVoid5/), or [good, top notch stuff](https://www.reddit.com/r/EmboldenTheE/)?

Arbitrary writing constraints and softer nudges can be interesting instruments of creativity.

Evaluators can be combined and weighted.

### [](#communities)Communities

i think it could be a really cool unique social experience

branching off of each other's writing

sharing in the constraints for creative writing with a consistency across the collaboration

a natural medium for nonlinear stories

I don't know what would work best for the granularity and structure, but it could have communities with independent moderation and rules, similar to Reddit, and pages with seed texts... and then maybe special rules could be applied to certain pages, possibly based on tags... and maybe some tags could only be set by moderators and some could be set by anyone...

A simpler model could be that there are just pages with seed text and rules set initially by a user.

Either way, there'd be links, of course.

### [](#long-form-writing)Long-form writing

An alternate (maybe not exclusive) direction for this project to take could be towards longer-form writing like articles and even books. In that case it should work inline, and it would be more focused on _editing_, rather than pure exploration. I think this is less interesting for now.

### [](#see-also)See also

*   a different textploration idea: [Intimate Codex](https://andymakes.itch.io/intimate-codex)
*   the [LibraryOfBabel](https://www.reddit.com/r/LibraryofBabel/) subreddit
*   the [LibraryOfBabelCollaboration](https://docs.google.com/document/d/1OFGbgKawAdHdWvf8NQgXijwcW77iE1B5IIi9dwveiPs/edit?usp=sharing) \- a google doc you're free to edit
*   [Your World of Text](https://www.yourworldoftext.com/) and [Our World of Text](https://ourworldoftext.com/)
*   [Dasher](http://www.bltt.org/software/dasher/), a text input method where you zoom and drill down into letters, weighted by how likely it thinks you are to want to use them from context
*   [rnn-writer](https://github.com/robinsloan/rnn-writer), a sentence autocompleter, which could be integrated into this project, via their [torch-rnn-server](https://github.com/robinsloan/torch-rnn-server), or even their hosted [text.bargains](https://text.bargains)
*   [gpt-2](https://github.com/openai/gpt-2), a newer text autocompleter

[](#development)Development
---------------------------

I'm using [Create React App](https://github.com/facebook/create-react-app) for the webpack + dev server setup, even tho i'm not using React.

*   [Clone the repo](https://help.github.com/en/articles/cloning-a-repository)
*   Install dependencies with `yarn install`

In the project directory, you can run:

### [](#npm-start)`npm start`

Runs the app in the development mode.  
Open [http://localhost:3000](http://localhost:3000) to view it in the browser.

The page will reload if you make edits.  
You will also see any lint errors in the console.

### [](#npm-test)`npm test`

Launches the test runner in the interactive watch mode.  
See the docs about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.

### [](#npm-run-build)`npm run build`

Builds the app for production to the `build` folder.

The build is minified and the filenames include the hashes.

### [](#npm-run-deploy)`npm run deploy`

Builds, and then pushes to `gh-pages`

[](#license)License
-------------------

MIT licensed. See [LICENSE.txt](/1j01/diverge/blob/master/LICENSE.txt) for details.

set Restriction 

*   © 2019 GitHub, Inc.
*   [Terms](https://github.com/site/terms)
*   [Privacy](https://github.com/site/privacy)
*   [Security](https://github.com/security)
*   [Status](https://githubstatus.com/)
*   [Help](https://help.github.com)

[](https://github.com "GitHub")

*   [Contact GitHub](https://github.com/contact)
*   [Pricing](https://github.com/pricing)
*   [API](https://developer.github.com)
*   [Training](https://training.github.com)
*   [Blog](https://github.blog)
*   [About](https://github.com/about)

You can’t perform that action at this time.

You signed in with another tab or window. Reload to refresh your session. You signed out in another tab or window. Reload to refresh your session.

![](blob:https://github.com/905a7fbd-fe4f-4b64-945d-dcfe12757e3f)

![](chrome-extension://jgnennkfpahpjpbmbbodaipgoilccmco/img/icon128.png)

_navigate_next_

_close_

_close_Close

_open\_in\_new_Open in new tab

Add Tags ?

articles

blockstack

bookmarks

dapp

github

interesting

learn

music

paste

swedish

AddDeleteNo

close

close

Add Tags ?

AddDeleteNo

close

close
